cmd/libsnap-confine-private/cgroup-support.c

```diff
@@ -66,3 +67,18 @@ void sc_cgroup_create_and_join(const char *parent, const char *name, pid_t pid)
     }
     debug("moved process %ld to cgroup hierarchy %s/%s", (long)pid, parent, name);
 }
+
+static const char *cgroup_dir = "/sys/fs/cgroup";
+// from statfs(2)
+static const int CGROUP2_SUPER_MAGIC =  0x63677270;
```

Comment:
I assume that this isn't inherited from say `include/uapi/linux/magic.h` because it's not available on all the systems we need to compile on and also because we don't want to use external dependencies at all in snap-confine?

I see that this magic number is defined in include/uapi/linux/magic.h on bionic and disco at least (in `cat /usr/src/linux-headers-$(uname -r)/include/uapi/linux/magic.h  | grep CGROUP2`), but notably not on xenial. From my experience I find it more maintainable to inherit magic values like these from an upstream source whenever I can, but I understand if that's not desirable for snap-confine specifically.

Reply:
FWIW the value is correct https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/magic.h#L62